src/etc/inc/plugins.inc.d/dnsmasq.inc

```diff
@@ -103,102 +145,12 @@ function dnsmasq_configure_do($verbose = false)
-    if (isset($config['dnsmasq']['port']) && is_port($config['dnsmasq']['port'])) {
-        $args .= " --port={$config['dnsmasq']['port']} ";
-    }
-
-    if (!empty($config['dnsmasq']['interface'])) {
-        $ifs = [];
-        foreach (explode(',', $config['dnsmasq']['interface']) as $ifname) {
-            if (!empty($config['interfaces'][$ifname]) && !empty($config['interfaces'][$ifname]['if'])) {
-                $ifs[] = $config['interfaces'][$ifname]['if'];
-            }
-        }
-        $args .= " --interface=" . implode(',', $ifs) . " ";
-
-        if (!empty($addresses) && !empty($config['dnsmasq']['strictbind'])) {
-            $args .= ' --bind-interfaces ';
-        }
-    }
-
-    if (!empty($config['dnsmasq']['no_private_reverse'])) {
-        $args .= ' --bogus-priv ';
-    }
-
-    foreach (config_read_array('dnsmasq', 'domainoverrides') as $override) {
-        $ip = $override['ip'];
-        if (!empty($ip) && !empty($override['port'])) {
-            $ip .= '#' . $override['port'];
-        }
-        if (!empty($ip) && !empty($override['srcip'])) {
-            $ip .= '@' . $override['srcip'];
-        }
-
-        $args .= ' --server=' . escapeshellarg('/' . $override['domain'] . '/' . $ip);
-
-        if (!isset($config['system']['webgui']['nodnsrebindcheck'])) {
-            $args .= ' --rebind-domain-ok=' . escapeshellarg('/' . $override['domain'] . '/') . ' ';
-        }
-    }
-
-    if (!empty($config['dnsmasq']['strict_order'])) {
-        $args .= ' --strict-order ';
-    } else {
-        $args .= ' --all-servers ';
-    }
-
-    if (!empty($config['dnsmasq']['domain_needed'])) {
-        $args .= ' --domain-needed ';
-    }
-
-    if (!empty($config['dnsmasq']['dnssec'])) {
-        $args .= ' --dnssec ';
-        $args .= ' --trust-anchor=.,20326,8,2,E06D44B80B8F1D39A95C0B0D7C65D08458E880409BBC683457104237C7F8EC8D ';
-        $args .= ' --trust-anchor=.,38696,8,2,683D2D0ACB8C9B712A1948B27F741219298D0A450D612C483AF444A4C0FB2B16 ';
-    }
-
-    if (!empty($config['dnsmasq']['log_queries'])) {
-        $args .= ' --log-queries=extra ';
-    }
-
-    if (!empty($config['dnsmasq']['no_hosts'])) {
-        $args .= ' --no-hosts ';
-    }
-
-    if (!empty($config['dnsmasq']['dns_forward_max'])) {
-        $args .= " --dns-forward-max={$config['dnsmasq']['dns_forward_max']} ";
-    } else {
-        $args .= ' --dns-forward-max=5000 ';
-    }
-
-    if (!empty($config['dnsmasq']['cache_size'])) {
-        $args .= " --cache-size={$config['dnsmasq']['cache_size']} ";
-    } else {
-        $args .= ' --cache-size=10000 ';
-    }
-
-    if (!empty($config['dnsmasq']['local_ttl'])) {
-        $args .= " --local-ttl={$config['dnsmasq']['local_ttl']} ";
-    } else {
-        $args .= ' --local-ttl=1 ';
-    }
-
-    $args .= ' --conf-dir=/usr/local/etc/dnsmasq.conf.d,\*.conf ';
-
     _dnsmasq_add_host_entries();
 
-    mwexec("/usr/local/sbin/dnsmasq {$args}");
+    mwexec("/usr/local/sbin/dnsmasq");
```

Comment:
can we consider rc.d scripting?

Reply:
we have tricks for that: 

    dnsmasq_skip="YES"

and we keep steering through rc.d conf files but keep some of the legacy glue... it solves the pid vs disable issue BTW